pkg/lib/cockpit.d.ts

```diff
@@ -192,11 +192,29 @@ declare module 'cockpit' {
 
     /* === String helpers ======================== */
 
+    type FormatOptions = {
+        precision?: number;
+        separate?: boolean;
+    };
+
+    type ByteUnit = {
+        name: string | null;
+        factor: number;
+        selected?: boolean;
+    };
+
     function gettext(message: string): string;
     function gettext(context: string, message?: string): string;
     function ngettext(message1: string, messageN: string, n: number): string;
     function ngettext(context: string, message1: string, messageN: string, n: number): string;
 
-    function format_bytes(n: number): string;
     function format(format_string: string, ...args: unknown[]): string;
+    function format_number(n: number, precision?: number): string
+    function format_bytes(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
+    function format_bytes(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate: true }): string[];
+    function format_bytes_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
+    function format_bytes_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate: true }): string[];
+    function format_bits_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { seperate?: false }): string;
+    function format_bits_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate: true }): string[];
```

Comment:
Same thing as I mentioned in the spawn PR: you really want `options: FormatOptions & { separate: true }` here.  If you use `options?:` then this possibility will match even if `options` is not given...

Reply:
Hmmmm I think I am misunderstanding how typescript works. `options` is optional. Looking at the file type, I need to also provide a non-optional one?

```
    function file(
        path: string,
        options?: FileOpenOptions & { binary?: false; syntax?: undefined; }
    ): FileHandle<string>;
    function file(
        path: string,
        options: FileOpenOptions & { binary: true; syntax?: undefined; }
    ): FileHandle<Uint8Array>;
```

Reply:
Ah! So if binary == true, options is not optional well that makes sense :)